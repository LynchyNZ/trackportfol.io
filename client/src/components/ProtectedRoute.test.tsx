import React from 'react';
import { shallow } from 'enzyme';
import ProtectedRouteComponent, { ProtectedRouteProps } from './ProtectedRoute';

const defaultProtectedRouteProps: ProtectedRouteProps = {
  authenticationPath: '/login',
};

const ProtectedApp = () => {
  return <div>Private page</div>;
};

describe('ProtectedRoute', () => {
  it('renders the ProtectedRoute component', () => {
    expect(ProtectedRouteComponent).toMatchSnapshot();
  });

  it('does not display the component when not logged in', () => {
    const result = shallow(
      <ProtectedRouteComponent path='/test' component={ProtectedApp} {...defaultProtectedRouteProps} />
    ).contains(<ProtectedApp />);
    expect(result).toBeFalsy();
  });

  it('does not display the component contents when not logged in', () => {
    const result = shallow(
      <ProtectedRouteComponent path='/test' component={ProtectedApp} {...defaultProtectedRouteProps} />
    ).contains('Private page');
    expect(result).toBeFalsy();
  });
});
