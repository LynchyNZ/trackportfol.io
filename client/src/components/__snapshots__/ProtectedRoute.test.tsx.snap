// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute renders the ProtectedRoute component 1`] = `[Function]`;
